definitions/npm/react-router-dom_v6.x.x/flow_v0.201.x-/react-router-dom_v6.x.x.js

```diff
@@ -226,6 +226,245 @@ declare module 'react-router-dom' {
+      formEncType: FormEncType,
+      formData: FormData,
+      data: TData | void,
+      ' _hasFetcherDoneAnything '?: boolean,
+    },
+  };
+
+  declare export type Fetcher<TData = any> = FetcherStates<TData>[$Keys<
+    FetcherStates<TData>
+  >];
+
+  declare type BlockerBlocked = {|
+    state: 'blocked',
+    reset(): void,
+    proceed(): void,
+    location: Location,
+  |};
+
+  declare type BlockerUnblocked = {|
+    state: 'unblocked',
+    reset: void,
+    proceed: void,
+    location: void,
+  |};
+
+  declare type BlockerProceeding = {|
+    state: 'proceeding',
+    reset: void,
+    proceed: void,
+    location: Location,
+  |};
+
+  declare export type Blocker =
+    | BlockerUnblocked
+    | BlockerBlocked
+    | BlockerProceeding;
+
+  /**
+   * State maintained internally by the router.  During a navigation, all states
+   * reflect the the "old" location unless otherwise noted.
+   */
+  declare export type RouterState = {|
+    /**
+     * The action of the most recent navigation
+     */
+    historyAction: HistoryAction,
+
+    /**
+     * The current location reflected by the router
+     */
+    location: Location,
+
+    /**
+     * The current set of route matches
+     */
+    matches: AgnosticDataRouteMatch[],
+
+    /**
+     * Tracks whether we've completed our initial data load
+     */
+    initialized: boolean,
+
+    /**
+     * Current scroll position we should start at for a new view
+     *  - number -> scroll position to restore to
+     *  - false -> do not restore scroll at all (used during submissions)
+     *  - null -> don't have a saved position, scroll to hash or top of page
+     */
+    restoreScrollPosition: number | false | null,
+
+    /**
+     * Indicate whether this navigation should skip resetting the scroll position
+     * if we are unable to restore the scroll position
+     */
+    preventScrollReset: boolean,
+
+    /**
+     * Tracks the state of the current navigation
+     */
+    navigation: Navigation,
+
+    /**
+     * Tracks any in-progress revalidations
+     */
+    revalidation: RevalidationState,
+
+    /**
+     * Data from the loaders for the current matches
+     */
+    loaderData: RouteData,
+
+    /**
+     * Data from the action for the current matches
+     */
+    actionData: RouteData | null,
+
+    /**
+     * Errors caught from loaders for the current matches
+     */
+    errors: RouteData | null,
+
+    /**
+     * Map of current fetchers
+     */
+    fetchers: Map<string, Fetcher<>>,
+
+    /**
+     * Map of current blockers
+     */
+    blockers: Map<string, Blocker>,
+  |};
+
+  /**
+   * Data that can be passed into hydrate a Router from SSR
+   */
+  declare export type HydrationState = Partial<{|
+    loaderData: RouterState['loaderData'],
+    actionData: RouterState['actionData'],
+    errors: RouterState['errors'],
+  |}>;
+
+  declare export type RelativeRoutingType = 'route' | 'path';
+
+  declare type BaseNavigateOptions = {
```

Comment:
why is this type declared privately, never exported, and reexported as a different name? Could it just be called `LinkNavigateOptions` to keep it simple

Reply:
Like the rest of this PR, I tried to be the most close to typescript definition 
https://github.com/remix-run/react-router/blob/edf1dfc3327f721045ed672870b4f99825aa8fb8/packages/router/router.ts#L432

Reply:
I guess we can keep it this way. Maybe there's a reason for it that we'll need in the future if you copy future types